### DISCUSSION TITLE
Out of the box snippets misconfigured?

### DISCUSSION THREAD

**Discussion by jpgeek** [LazyVim/LazyVim#4608] [Q&A] [answered]
I am using the default configuration of LazyVim with only ruby and luasnip enabled, both from :LazyExtras.  Typing the following in a ruby file:

```
def foo
  if a == b
```

After 'foo', the snippet dropdown appears with a single suggestion of something like "foo = Text".  Hitting Enter once selects that and hitting it again enters a CR.  After entering in the "b", it pops up a list of things, starting with a snipped for something like "a = variable { || ... }.  Hitting enter here injects the snippet (not what I want), so I have to hit escape, then i, then enter just to get a CR entered.  This adds 3 keystrokes to the two trivial lines above.  I have reinstalled several times but get the same experience.  

Is this really the intended functionality?  I can't imagine anyone would want this.  Everything else in LazyVim is super polished so I suspect I have to be doing something very wrong.

I have added the following to prevent the autocomplete and change the mapping and it works fine.  I still feel like I must be missing something fundamental though.  Any help would be greatly appreciated, even if it is just to say "yup, that's how it is supposed to be".

```
# ~/.config/nvim/lua/plugins/cmp.lua
return {
  "hrsh7th/nvim-cmp",
  opts = function(_, opts)
    local cmp = require("cmp")
    opts.completion = {
      autocomplete = false, -- Prevent the menu from showing up automatically
    }
    opts.mapping = cmp.mapping.preset.insert({
      ["<C-s>"] = cmp.mapping.complete(), -- Trigger completion manually with <C-Space>
      ["<CR>"] = cmp.mapping.confirm({
        behavior = cmp.ConfirmBehavior.Replace,
        select = false,
      }),
      -- Other mappings...
    })
  end,
}
```

**Comment by dpetka2001** [comment, marked as answer]
Yes, the mappings are default Lazyvim behavior and users can change that in their personal configuration. `<CR>` is used for selecting nvim-cmp entries and ` <C-CR>` to actually insert a new line (so you don't have to hit `<esc>` and then `i` and then `<cr>` again) . The menu popping up is default nvim-cmp behavior and that can also be changed like you did. You're talking about personal preferences here, so any changes go to your personal configuration. Naturally, everyone has different habits and preferences and that's why the personal configuration exists. In my personal configuration, i added a setting to not explicitly select an entry unless i choose one with moving up/down because that's how i personally prefer it.

**Reply by jpgeek** [reply]
@dpetka2001 , thank you kindly for the quick and insightful answer!   

I am indeed talking about personal preferences (and my own ignorance on this point).  I did not see the <C-CR> binding - that makes the default setup more useful.  Your configuration sounds nice too - I may give that a try.  Something like this?

```
return {
  "hrsh7th/nvim-cmp",
  opts = function(_, opts)
    local cmp = require("cmp")

    opts.mapping = cmp.mapping.preset.insert({
      ["<CR>"] = cmp.mapping.confirm({
        behavior = cmp.ConfirmBehavior.Replace,
        select = false,  -- Only confirm if a completion is explicitly selected
      }),
    })
  end,
}

```

**Reply by dpetka2001** [reply]
```lua
      opts.preselect = cmp.PreselectMode.None
      opts.completion = {
        completeopt = "menu,menuone,noinsert,noselect",
        -- autocomplete = false,
      }
```
I added this to disable preselection. Test it out and keep it or discard it :)

**Reply by jpgeek** [reply]
Very nice!  More concise and easier to use than what I had.
